cli_ftd/user/sensor/stm32/stm32_i2c.c

```diff
@@ -185,40 +282,167 @@ bool stm32__read_data(uint8_t write_address, uint8_t readBuf[],
 
   bool ret = I2C_transfer(handle, &transaction);
   if (!ret) {
-    // I2C_STATUS_SUCCESS;
-    debugPrintf("I2C Unsuccessful transfer\r\n");
+    i2c_debugPrintf("I2C Unsuccessful transfer\r\n");
     I2C_close(handle);
     return false;
   }
-  debugPrintf("Status: %d\r\n", transaction.status);
+  // i2c_debugPrintf("Status: %d\r\n", transaction.status);
 
   I2C_close(handle);
   return true;
 }
 
-// Static functions
-static uint32_t stm32__convert_to_uint32(uint8_t *data) {
+uint32_t stm32__convert_to_uint32(uint8_t *data) {
   uint32_t rdata =
       ((data[3] << 24) | (data[2] << 16) | (data[1] << 8) | data[0]);
   ;
-  debugPrintf("%lx\r\n", rdata);
+  // debugPrintf("%lx\r\n", rdata);
   return rdata;
 }
 
-static uint16_t stm32__convert_to_uint16(uint8_t *data) {
+uint16_t stm32__convert_to_uint16(uint8_t *data) {
   uint16_t rdata = ((data[1] << 8) | data[0]);
-  debugPrintf("%x\r\n", rdata);
+  // debugPrintf("%x\r\n", rdata);
   return rdata;
 }
 
-static float stm32__convert_uint32_float_structure_to_float(uint32_t data) {
+float stm32__convert_uint32_float_structure_to_float(uint32_t data) {
   float *rdata = (float *)&data;
 
 #if DEBUG_PRINT
   char buffer[32] = {0};
   float__convert_float_to_string((*rdata), buffer);
-  debugPrintf("%s\r\n", buffer);
+  // debugPrintf("%s\r\n", buffer);
 #endif
 
   return (*rdata);
 }
+
+void convert_float_and_print(float data, const char *identifier) {
+  char buffer[12] = {0};
+  float__convert_float_to_string(data, buffer);
+  debugPrintf("%s: %s\r\n", identifier, buffer);
+}
+
+void uint32_print(uint32_t data, const char *identifier) {
+  debugPrintf("%s: %lu\r\n", identifier, data);
+}
+
+void i32_3_print(int32_t data[], const char *identifier) {
+  debugPrintf("%s: %ld %ld %ld\r\n", identifier, data[0], data[1], data[2]);
+  memset(data, 0, 3 * sizeof(int32_t));
+}
+
+// STM Task-related code
+
+/* Application thread call stack */
+static char stm_i2c_comm_stack[TASK_CONFIG_STM_TASK_STACK_SIZE];
+
+// stm_i2c_comm Task
+
+void stm_i2c_comm_taskCreate(void) {
```

Comment:
shift this to its own source and header file
For example create in user/sensor/stm32/stm32_i2c_task.c/.h etc

Reply:
Done. Updated in new PR.